### PR TITLE
feat: add linters dependencies

### PR DESCRIPTION
**Pull request by Arcle1**
- Add linters dependencies

### CODE REVIEW

File: script/gulp/task.build.js [hunk 1]
```diff
@@ -66,20 +86,60 @@ const getMeta = (file) => {
...
 
+const getDeps = (file) => {
+    const packages = bufferToJson(file.contents);
+    const manager = registry.getManager(packages.manager);
+    return Promise.all(
+        packages.linters.map(async (lint) => {
+            const versions = await getVersions(lint, manager);
+            if (versions.length === 0) {
+                const deps = await manager.getDeps(lint.name);
+                const versions = await manager.getVersions(lint.name);
```
Comment (svoboda-rabstvo) [inline script/gulp/task.build.js]: It's don't need

File: script/gulp/task.build.js [hunk 2]
```diff
@@ -66,20 +86,60 @@ const getMeta = (file) => {
...
  * @return {array} obj - array of objects with data of linters
  */
-const getFiles = () => {
+const getDepsFiles = () => {
     return through2.obj(async function (file, enc, next) {
 
-        let metaLinters = await getMeta(file);
-        metaLinters.map(obj => {
+        let depsLinters = await getDeps(file);
+        console.log(depsLinters);
```
Comment (svoboda-rabstvo) [inline script/gulp/task.build.js]: `console.log` need only for debug. Please, remove it

File: script/gulp/task.build.js [hunk 3]
```diff
@@ -66,20 +86,60 @@ const getMeta = (file) => {
...
+const getDeps = (file) => {
+    const packages = bufferToJson(file.contents);
+    const manager = registry.getManager(packages.manager);
+    return Promise.all(
+        packages.linters.map(async (lint) => {
+            const versions = await getVersions(lint, manager);
+            if (versions.length === 0) {
+                const deps = await manager.getDeps(lint.name);
+                const versions = await manager.getVersions(lint.name);
+                const lastVersion = versions[versions.length - 1];
```
Comment (svoboda-rabstvo) [inline script/gulp/task.build.js]: It's dont' need too

File: script/gulp/task.build.js [hunk 4]
```diff
@@ -66,20 +86,60 @@ const getMeta = (file) => {
...
+        next();
+    });
+};
+
 /**
  * Get array of objects, where each contains meta of linter and base & path where it will be stored
  * @return {array} obj - array of objects with data of linters
  */
-const getFiles = () => {
+const getDepsFiles = () => {
```
Comment (svoboda-rabstvo) [inline script/gulp/task.build.js]: Please, combine `getDepsFiles` and `getPackagesFiles` functions

File: script/gulp/task.build.js (hunk 3 above)
Comment (Arcle1) [reply script/gulp/task.build.js]: @svoboda-rabstvo we need it to write in folder with version number.

File: script/gulp/task.build.js (hunk 1 above)
Comment (Arcle1) [reply script/gulp/task.build.js]: @svoboda-rabstvo done.

File: script/gulp/index.js [hunk 5]
```diff
@@ -19,7 +19,72 @@ const amd = {
...
+const getFiles = (func, fileName, manager) => {
+    if (manager !== null) manager = amd.registry.getManager(manager);
+
+    return amd.through2.obj(async function (file, enc, next) {
+        const packages = fnc.bufferToJson(file.contents);
+        if (manager === null) manager = amd.registry.getManager(packages.manager);
+        const dataLinters = await func(manager, packages);
+
+        dataLinters.map(obj => {
+            if (!Array.isArray(obj)) {
```
Comment (svoboda-rabstvo) [inline script/gulp/index.js]: I think we don't need to check `isArray`

File: script/gulp/index.js [hunk 6]
```diff
@@ -19,7 +19,72 @@ const amd = {
...
+ * @param {string} fileName - name of file to stored data
+ * @param {string} manager - manager
+ * @return {array} obj - array of objects with data of linters
+ */
+const getFiles = (func, fileName, manager) => {
+    if (manager !== null) manager = amd.registry.getManager(manager);
+
+    return amd.through2.obj(async function (file, enc, next) {
+        const packages = fnc.bufferToJson(file.contents);
+        if (manager === null) manager = amd.registry.getManager(packages.manager);
```
Comment (svoboda-rabstvo) [inline script/gulp/index.js]: Use ternary operator

File: script/gulp/index.js [hunk 7]
```diff
@@ -19,7 +19,72 @@ const amd = {
...
+
+/**
+ * Write files to build
+ * @param {amd.file} file - file with content
+ * @param {boolean} version - write file to folder with version name
+ * @return {string}
+ */
+const writeFiles = (file, version) => {
+    const content = JSON.parse(fnc.bufferToJson(file.contents));
+    return version ? amd.path.join(cfg.build.dir, content.name, content.version) : amd.path.join(cfg.build.dir, content.name);
```
Comment (svoboda-rabstvo) [inline script/gulp/index.js]: Refactor this please

File: script/gulp/task.build.js [hunk 8]
```diff
@@ -31,6 +31,24 @@ const getPackageSchema = (meta) => {
...
 };
 
+/**
+ * Add in deps fields from schema such as schema uri, version and package uri
+ * @param {object} deps - object with linter deps
+ * @param {string} lintName - name of linter
+ * @param {string} version - version of linter
+ * @return {object} deps - object with linter deps & schema properties
+ */
+const getDepsSchema = (deps, lintName, version) => {
```
Comment (svoboda-rabstvo) [inline script/gulp/task.build.js]: This function need only two param: `deps` and `package`

File: script/gulp/task.deps.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+/**
+ * Get deps for all linters & all versions from linters file
+ * @param {string} manager - package manager (npm, pip, etc.)
+ * @param {object} lint - linter from build
+ * @return {array} - two-dimensional array with linters deps
+ */
+const getDeps = async (manager, lint) => {
+    if (typeof lint === 'string') lint = JSON.parse(lint);
```
Comment (svoboda-rabstvo) [inline script/gulp/task.deps.js]: I think we don't need it

File: script/gulp/task.package.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Add in meta fields from schema such as schema uri, version and package uri
+ * @param {object} meta - object with linter meta
+ * @return {object} meta - object with linter meta & schema properties
+ */
+const getPackageSchema = (meta) => {
+    return {
+        $schema: schema.package.$id,
+        $version: schema.package.$version,
+        package: `${meta.name}:${meta.version}`,
```
Comment (svoboda-rabstvo) [inline script/gulp/task.package.js]: Id type is  `uri`, e.g.: `https://hub.linterhub.com/alex/0.0.1/package.json`

**Pull request merged**